lms/djangoapps/courseware/access.py

```diff
@@ -370,17 +396,18 @@ def _can_load_course_overview(user, course_overview):
         The user doesn't have to be enrolled in the course in order to have load
         load access.
     """
-    return (
-        not course_overview.visible_to_staff_only
+    return _has_staff_access_to_descriptor(user, course_overview, course_overview.id) or \
+        (
+        _is_visible_to_staff_only(course_overview)
         and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
-    ) or _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
+        )
 
 
 _COURSE_OVERVIEW_CHECKERS = {
     'load': _can_load_course_overview,
     'load_mobile': lambda user, course_overview: (
-        _can_load_course_overview(user, course_overview)
-        and _can_load_course_on_mobile(user, course_overview)
+        _can_load_course_overview(user, course_overview) if not _can_load_course_overview(user, course_overview)
+        else _can_load_course_on_mobile(user, course_overview)
```

Comment:
the way `_dispatch` works, the values in the checkers dictionary have to be functions